src/neuronmusic/nrnmusic_dynam.cpp

```diff
@@ -0,0 +1,83 @@
+#include <../../nrnconf.h>
+#include <stdio.h>
+#include <assert.h>
+#include <errno.h>
+#include "oc_ansi.h"
+#include "nrnmusicapi.h"
+#include <array>
+#include <cstdlib>
+
+#include "nrnwrap_dlfcn.h"
+
+void (*p_nrnmusic_runtime_phase)();
+void (*p_nrnmusic_injectlist)(void*, double);
+void (*p_nrnmusic_init)(int* parg, char*** pargv);
+void (*p_nrnmusic_terminate)();
+
+#if DARWIN || defined(__linux__)
+extern const char* path_prefix_to_libnrniv();
+#else
+#error MUSIC dynamic loading not implemented for this architecture.
+#endif
+
+static void load_music() {
+    void* handle = nullptr;
+    using const_char_ptr = const char*;
+    // would be nice to figure out path from `which music` but not on
+    // every rank if nhost is large. So demand ...
+    const_char_ptr music_path = const_char_ptr{std::getenv("MUSIC_LIB_NRN_PATH")};
```

Comment:
if we are introducing a new env variable, we should prefer to have the `NRN` prefix at the beginning i.e. `NRN_MUSIC_LIB_PATH`?

Reply:
I agree. Will change to NRN_LIBMUSIC_PATH . Now that I think about it, we also have in nrnmpi_dynam.cpp the
variable ```"MPI_LIB_NRN_PATH"``` . I guess that can be changed in another PR along with, perhaps, the other environment variables that don't follow that prefix.